### DISCUSSION TITLE
bug: Can't connect to osu server

### DISCUSSION THREAD

**Discussion by mames1dev** [osuAkatsuki/bancho.py#403] [Q&A] [answered]
### Describe the bug

can't connect to server from my pc...
I'd like to know why I can't connect.

### To Reproduce

1. start up server.
2. can't connect to server from my pc (but from browser, i can access there.)
![image](https://user-images.githubusercontent.com/88868436/218289347-4e494e3c-53c6-419f-883d-5b856e2e66ce.png)
![image](https://user-images.githubusercontent.com/88868436/218289352-e98fa44b-61f6-411f-b47d-66c697ab4d8a.png)

### Expected behavior

i don't know... sorry

### bancho.py Version

maybe 3.2.2

### Python Version

3.9.X (Default)

### Relevant log output

```shell
root@localhost:~/bancho.py# make run
pipenv run python main.py
Loading .env environment variables...
[02:12:38AM] It is not recommended to run bancho.py as root, especially in production..
[02:12:38AM] Running the server with root privileges is not recommended.
[02:12:38AM] Fetching channels from sql.
[02:12:38AM] Fetching clans from sql.
[02:12:38AM] Fetching mappools from sql.
[02:12:38AM] Initializing housekeeping tasks.
[02:12:38AM] Startup process complete.
```

**Comment by mames1dev** [comment]
i used 'devserver' to connect server. 
ex: 'server_address':'port'

**Comment by yo-ru** [comment]
the startup variable "devserver" only works with domain names as far as I have tested.

It should look something like this:
`C:/path/to/osu!/osu!.exe -devserver example.com`

Please note that osu! requires https, so you will need to have a proper SSL configuration.

**Comment by mames1dev** [comment]
thanks for your advice! I tried that and it did not solve the problem. Do I need to set up SSL other than under image? (I rewrote it and set it up)
![image](https://user-images.githubusercontent.com/88868436/218293236-78650397-a90e-4ea8-a19f-68cbf920f1ae.png)
![image](https://user-images.githubusercontent.com/88868436/218293266-ba08742a-b42b-4bd0-8259-03ae9ede9fcb.png)
![image](https://user-images.githubusercontent.com/88868436/218293359-760dae36-0a39-4f5d-bad6-751f37b67a65.png)

**Comment by mames1dev** [comment]
my ssl status


![image](https://user-images.githubusercontent.com/88868436/218294413-24ca9d23-c409-4fdd-861b-c90b2e53e0ee.png)

**Comment by yo-ru** [comment]
Ideally:
Please make sure your nginx config is setup to listen on port 443 with your SSL certificate & key paths set correctly.

Alternatively:
If you want to go the easy route. Use Cloudflare with a the "Flexible" shield level and comment out `listen 443` and anything related to SSL. Cloudflare will handle your SSL certificate for you.

---
You should be able to access https://c.mamesosu.net without running into any SSL errors.

Your shortcut should look like this:
`C:/path/to/osu!/osu!.exe -devserver mamesosu.net`

Please note that ports are not supported in the "devserver" startup variable.

**Reply by mames1dev** [reply]
Thanks to many contributors, https://c.mamesosu.net can be accessed without ssl errors. However, the server cannot be connected from osu

**Reply by mames1dev** [reply]
maybe receiving requests from browsers

![image](https://user-images.githubusercontent.com/88868436/218333996-99d6c579-8a6a-4bc2-8660-365a0007c1b5.png)

**Comment by mames1dev** [comment]
Thanks for your reply :) 
I have changed the ssl settings and seem to be able to connect, but I get tons of errors.
If i launched osu, many http request error occured....
please teach me how to fix this ploblem..! thank you!

# error 
![image](https://user-images.githubusercontent.com/88868436/218300948-cc52c6cc-da34-49fe-aeca-0df955546570.png)

# cloudflare
![image](https://user-images.githubusercontent.com/88868436/218300941-782caca1-95ce-4834-b248-445126e7318c.png)
![image](https://user-images.githubusercontent.com/88868436/218301033-3e9bf9ad-7fe0-499b-8c55-f3fda3c25a5a.png)

# nginx
#NOTE: if you wish to only connect using fallback, you can
#remove all ssl related content from the configuration.

upstream bancho {
        server unix:/tmp/bancho.sock fail_timeout=0;
}

#c[e4]?.ppy.sh is used for bancho
#osu.ppy.sh is used for /web, /api, etc.
#a.ppy.sh is used for osu! avatars

server {
        listen 443 ssl;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name ~^(?:c[e4]?|osu|b|api)\.mamesosu.net$;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
        ssl_protocols TLSv1.2;

        location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Real-IP  $remote_addr;
                proxy_set_header Host $http_host;
                add_header Access-Control-Allow-Origin *;
                proxy_redirect off;
                proxy_pass http://bancho;
        }
}

server {
        listen 443 ssl;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name assets.mamesosu.net;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_protocols TLSv1.2;
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

        location / {
                default_type image/png;
                #XXX: you'll need to edit this to match your bancho.py .data/assets directory
                root bancho.py/.data/assets/;
        }
}

server {
        listen 443 ssl;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name a.mamesosu.net;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_protocols TLSv1.2;
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

        location / {
                root bancho.py/.data/avatars;
                try_files $uri $uri.png $uri.jpg $uri.gif $uri.jpeg $uri.jfif /default.jpg = 404;
        }
}
# nano.env
#bancho.py supports both INET and UNIX sockets
#NET: provide an IP for SERVER_ADDR, and a port for SERVER_PORT
#UNIX: provide a filesystem path for SERVER_ADDR, and no SERVER_PORT
SERVER_ADDR=i set global port 
SERVER_PORT=13381

DB_USER=user_name
DB_PASS=pass
DB_HOST=localhost
DB_PORT=port
DB_NAME=db_name

REDIS_USER=redis
REDIS_PASS=
REDIS_HOST=localhost
REDIS_PORT=port
REDIS_DB=0

OSU_API_KEY=api_key

DOMAIN=cmyui.xyz
MIRROR_URL=https://api.chimu.moe/v1

COMMAND_PREFIX=!

SEASONAL_BGS=https://akatsuki.pw/static/flower.png,https://i.cmyui.xyz/nrMT4V2RR3PR.jpeg

MENU_ICON_URL=https://akatsuki.pw/static/logos/logo_ingame.png
MENU_ONCLICK_URL=https://akatsuki.pw

DATADOG_API_KEY=
DATADOG_APP_KEY=

DEBUG=False

#redirect beatmaps, beatmapsets, and forum
#pages of maps to the official osu! website
REDIRECT_OSU_URLS=True

PP_CACHED_ACCS=90,95,98,99,100

DISALLOWED_NAMES=mrekk,vaxei,btmc,cookiezi
DISALLOWED_PASSWORDS=password,abc123
DISALLOW_OLD_CLIENTS=True

DISCORD_AUDIT_LOG_WEBHOOK=

#automatically share information with the primary
#developer of bancho.py (https://github.com/cmyui)
#for debugging & development purposes.
AUTOMATICALLY_REPORT_PROBLEMS=False

#advanced dev settings

##WARNING: only touch this once you've
##read through what it enables.
##you could put your server at risk.
DEVELOPER_MODE=False

**Comment by mames1dev** [comment]
from webpage

![image](https://user-images.githubusercontent.com/88868436/218301983-d6d46cc6-21d8-4687-b4be-abb9d6c86bc6.png)

**Comment by minisbett** [comment]
![image](https://user-images.githubusercontent.com/39670899/218306936-0ec2e934-a0d5-4b45-948d-fa1bc896bf13.png)
You need to change these to proxied. That way, the requests go through Cloudflares servers allowing for a flexible TLS encryption.

**Comment by minisbett** [comment]
![image](https://user-images.githubusercontent.com/39670899/218306993-4a5ff660-697f-43db-acff-6b93694a2a4c.png)
I also saw that you changed this and put your VPS IP in there. Revert that, it needs to have that socket path.

**Comment by minisbett** [comment]
![image](https://user-images.githubusercontent.com/39670899/218307040-4efe2c63-58e0-4988-baf6-c41a46c4b564.png)
You also need to put your domain here

**Comment by minisbett** [comment]
And you need to change all `listen 443 ssl` to `listen 80` as you were instructed to

**Comment by mames1dev** [comment]
thank you for your advice. i fixed these solution. but it wasn't solved....
What are other possible causes?　I'm sorry for troubling you. 

![image](https://user-images.githubusercontent.com/88868436/218309368-91cbcda5-38b8-4ca1-9d4c-511ff1f98822.png)
![image](https://user-images.githubusercontent.com/88868436/218309396-c703c872-e31a-4684-8661-515c0225bf1e.png)
![image](https://user-images.githubusercontent.com/88868436/218309535-15e60aea-e016-4009-9349-905e61bc1084.png)

**Comment by minisbett** [comment]
Could you share your nginx and .env (make sure to censor critical stuff like api keys and passwords) file again?

**Comment by mames1dev** [comment]
# nginx

  GNU nano 6.2                                                                                                                                                                                        /etc/nginx/sites-available/bancho.conf                                                                                                                                                                                                  
#NOTE: if you wish to only connect using fallback, you can
#remove all ssl related content from the configuration.

upstream bancho {
        server unix:/tmp/bancho.sock fail_timeout=0;
}

#c[e4]?.ppy.sh is used for bancho
#osu.ppy.sh is used for /web, /api, etc.
#a.ppy.sh is used for osu! avatars

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name ~^(?:c[e4]?|osu|b|api)\.mamesosu.net$;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
        ssl_protocols TLSv1.2;

        location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Real-IP  $remote_addr;
                proxy_set_header Host $http_host;
                add_header Access-Control-Allow-Origin *;
                proxy_redirect off;
                proxy_pass http://bancho;
        }
}

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name assets.mamesosu.net;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_protocols TLSv1.2;
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

        location / {
                default_type image/png;
                #XXX: you'll need to edit this to match your bancho.py .data/assets directory
                root bancho.py/.data/assets/;
        }
}

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name a.mamesosu.net;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_protocols TLSv1.2;
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

        location / {
                root bancho.py/.data/avatars;
                try_files $uri $uri.png $uri.jpg $uri.gif $uri.jpeg $uri.jfif /default.jpg = 404;
        }
}

# nano.env

#bancho.py supports both INET and UNIX sockets
#INET: provide an IP for SERVER_ADDR, and a port for SERVER_PORT
#UNIX: provide a filesystem path for SERVER_ADDR, and no SERVER_PORT
SERVER_ADDR=/tmp/bancho.sock
SERVER_PORT=

DB_USER=user
DB_PASS=pass
DB_HOST=localhost
DB_PORT=port
DB_NAME=dbname

REDIS_USER=redis
REDIS_PASS=
REDIS_HOST=localhost
REDIS_PORT=port
REDIS_DB=0

OSU_API_KEY=apikey

DOMAIN=mamesosu.net
MIRROR_URL=https://api.chimu.moe/v1

COMMAND_PREFIX=!

SEASONAL_BGS=https://akatsuki.pw/static/flower.png,https://i.cmyui.xyz/nrMT4V2RR3PR.jpeg

MENU_ICON_URL=https://akatsuki.pw/static/logos/logo_ingame.png
MENU_ONCLICK_URL=https://akatsuki.pw

DATADOG_API_KEY=
DATADOG_APP_KEY=

DEBUG=False

#edirect beatmaps, beatmapsets, and forum
#pages of maps to the official osu! website
REDIRECT_OSU_URLS=True

PP_CACHED_ACCS=90,95,98,99,100

DISALLOWED_NAMES=mrekk,vaxei,btmc,cookiezi
DISALLOWED_PASSWORDS=password,abc123
DISALLOW_OLD_CLIENTS=True

DISCORD_AUDIT_LOG_WEBHOOK=

#automatically share information with the primary
#developer of bancho.py (https://github.com/cmyui)
#for debugging & development purposes.
AUTOMATICALLY_REPORT_PROBLEMS=False

#advanced dev settings

##WARNING: only touch this once you've
##read through what it enables.
##you could put your server at risk.
DEVELOPER_MODE=False

these are env & nginx file

**Comment by minisbett** [comment]
![image](https://user-images.githubusercontent.com/39670899/218309998-18a6457f-295f-475a-8a90-601827822bbe.png)
Please make sure this is a valid path. e.g.: `/home/mames1dev/bancho.py/.data/assets/`

**Comment by mames1dev** [comment]
![image](https://user-images.githubusercontent.com/88868436/218310136-3dc8bcd6-db34-4fb5-a06b-a0b8d3bb2637.png)

I think it's a valid path...

**Comment by 7ez** [comment]
> ![image](https://user-images.githubusercontent.com/88868436/218310136-3dc8bcd6-db34-4fb5-a06b-a0b8d3bb2637.png)
> 
> I think it's a valid path...

 /root/bancho.py/.data/assets

**Comment by mames1dev** [comment]
sorry, It was my mistake. i changed these path. but No change was found.

# nginx
 GNU nano 6.2                                                                                                                                             /etc/nginx/sites-available/bancho.conf                                                                                                                                                       
#NOTE: if you wish to only connect using fallback, you can
#remove all ssl related content from the configuration.

upstream bancho {
        server unix:/tmp/bancho.sock fail_timeout=0;
}

#c[e4]?.ppy.sh is used for bancho
#osu.ppy.sh is used for /web, /api, etc.
#a.ppy.sh is used for osu! avatars

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name ~^(?:c[e4]?|osu|b|api)\.mamesosu.net$;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
        ssl_protocols TLSv1.2;

        location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Real-IP  $remote_addr;
                proxy_set_header Host $http_host;
                add_header Access-Control-Allow-Origin *;
                proxy_redirect off;
                proxy_pass http://bancho;
        }
}

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name assets.mamesosu.net;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_protocols TLSv1.2;
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

        location / {
                default_type image/png;
                # XXX: you'll need to edit this to match your bancho.py .data/assets directory
                root /root/bancho.py/.data/assets;
        }
}

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name a.mamesosu.net;

        #XXX: you'll need to edit this to match your certificate and key
        ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;

        #TODO: further ssl configuration
        ssl_protocols TLSv1.2;
        ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";

        location / {
                root root/bancho.py/.data/avatars;
                try_files $uri $uri.png $uri.jpg $uri.gif $uri.jpeg $uri.jfif /default.jpg = 404;
        }
}

**Comment by minisbett** [comment]
![image](https://user-images.githubusercontent.com/39670899/218310648-86fb457a-7c93-4d0f-9d08-aa417fde96c6.png)
Try to remove all of this SSL stuff, maybe it confuses nginx

**Comment by 7ez** [comment]
> sorry, It was my mistake. i changed these path. but No change was found.
> # nginx
> 
> GNU nano 6.2 /etc/nginx/sites-available/bancho.conf #NOTE: if you wish to only connect using fallback, you can #remove all ssl related content from the configuration.
> 
> upstream bancho { server unix:/tmp/bancho.sock fail_timeout=0; }
> 
> #c[e4]?.ppy.sh is used for bancho #osu.ppy.sh is used for /web, /api, etc. #a.ppy.sh is used for osu! avatars
> 
> server { listen 80;
> 
> ```
>     #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
>     server_name ~^(?:c[e4]?|osu|b|api)\.mamesosu.net$;
> 
>     #XXX: you'll need to edit this to match your certificate and key
>     ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
>     ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;
> 
>     #TODO: further ssl configuration
>     ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
>     ssl_protocols TLSv1.2;
> 
>     location / {
>             proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
>             proxy_set_header X-Real-IP  $remote_addr;
>             proxy_set_header Host $http_host;
>             add_header Access-Control-Allow-Origin *;
>             proxy_redirect off;
>             proxy_pass http://bancho;
>     }
> ```
> 
> }
> 
> server { listen 80;
> 
> ```
>     #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
>     server_name assets.mamesosu.net;
> 
>     #XXX: you'll need to edit this to match your certificate and key
>     ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
>     ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;
> 
>     #TODO: further ssl configuration
>     ssl_protocols TLSv1.2;
>     ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
> 
>     location / {
>             default_type image/png;
>             # XXX: you'll need to edit this to match your bancho.py .data/assets directory
>             root /root/bancho.py/.data/assets;
>     }
> ```
> 
> }
> 
> server { listen 80;
> 
> ```
>     #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
>     server_name a.mamesosu.net;
> 
>     #XXX: you'll need to edit this to match your certificate and key
>     ssl_certificate     /etc/letsencrypt/live/mamesosu.net/fullchain.pem;
>     ssl_certificate_key /etc/letsencrypt/live/mamesosu.net/privkey.pem;
> 
>     #TODO: further ssl configuration
>     ssl_protocols TLSv1.2;
>     ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH:@SECLEVEL=1";
> 
>     location / {
>             root root/bancho.py/.data/avatars;
>             try_files $uri $uri.png $uri.jpg $uri.gif $uri.jpeg $uri.jfif /default.jpg = 404;
>     }
> ```
> 
> }

![image](https://user-images.githubusercontent.com/36964527/218310682-ee4644df-c21a-4d24-a091-1f79b5ce6325.png)
you forgot a / at the beginning, also change `user www-data` into `user root` in /etc/nginx/nginx.conf

**Comment by mames1dev** [comment]
thanks for your advice. i fixed these solution. but i can't connect to the server from my pc... 
I'll paste the edited file just in case.

#  /etc/nginx/nginx.conf

user root;
worker_processes auto;
pid /run/nginx.pid;
include /etc/nginx/modules-enabled/*.conf;

events {
        worker_connections 768;
        #multi_accept on;
}

http {

        ##
        #Basic Settings
        ##

        sendfile on;
        tcp_nopush on;
        types_hash_max_size 2048;
        #server_tokens off;

        #server_names_hash_bucket_size 64;
        #server_name_in_redirect off;

        include /etc/nginx/mime.types;
        default_type application/octet-stream;

        ##
        #SSL Settings
        ##

        ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3; # Dropping SSLv3, ref: POODLE
        ssl_prefer_server_ciphers on;
     

   ##
        #Logging Settings
        ##

        access_log /var/log/nginx/access.log;
        error_log /var/log/nginx/error.log;

        ##
        #Gzip Settings
        ##

        gzip on;

        #gzip_vary on;
        #gzip_proxied any;
        #gzip_comp_level 6;
        #gzip_buffers 16 8k;
        #gzip_http_version 1.1;
        #gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;

        ##
        #Virtual Host Configs
        ##

        include /etc/nginx/conf.d/*.conf;
        include /etc/nginx/sites-enabled/*;
}


#mail {
       #See sample authentication script at:
       #http://wiki.nginx.org/ImapAuthenticateWithApachePhpScript

       #auth_http localhost/auth.php;
       #pop3_capabilities "TOP" "USER";
       #map_capabilities "IMAP4rev1" "UIDPLUS";

       server {
               listen     localhost:110;
               protocol   pop3;
               proxy      on;
       }

       server {
              listen     localhost:143;
               protocol   imap;
               proxy      on;
       }
}

# nginx

upstream bancho {
        server unix:/tmp/bancho.sock fail_timeout=0;
}

#c[e4]?.ppy.sh is used for bancho
#osu.ppy.sh is used for /web, /api, etc.
#a.ppy.sh is used for osu! avatars

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name ~^(?:c[e4]?|osu|b|api)\.mamesosu.net$;

        location / {
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Real-IP  $remote_addr;
                proxy_set_header Host $http_host;
                add_header Access-Control-Allow-Origin *;
                proxy_redirect off;
                proxy_pass http://bancho;
        }
}

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name assets.mamesosu.net;

        location / {
                default_type image/png;
                #XXX: you'll need to edit this to match your bancho.py .data/assets directory
                root /root/bancho.py/.data/assets;
        }
}

server {
        listen 80;

        #XXX: you'll need to edit this to match your domain (mine's cmyui.xyz)
        server_name a.mamesosu.net;

        location / {
                root /root/bancho.py/.data/avatars;
                try_files $uri $uri.png $uri.jpg $uri.gif $uri.jpeg $uri.jfif /default.jpg = 404;
        }
}

**Comment by mames1dev** [comment]
Other contributors have told me that the certificate or cloudflare settings are bad. Are those possibilities?

![image](https://user-images.githubusercontent.com/88868436/218313607-68dde892-893f-486e-b29a-c70fe7c98a90.png)

**Comment by mames1dev** [comment, marked as answer]
I have successfully connected after setting up the fine details of cloudflare. Thank you all very much!

![image](https://user-images.githubusercontent.com/88868436/218335996-e65b5c9c-644e-494d-8e28-43b612629bc8.png)
